mlx/primitives.cpp

```diff
@@ -2139,6 +2140,15 @@ std::vector<array> Scatter::vjp(
   const array& updates = primals.back();
   const std::vector<array> indices(primals.begin() + 1, primals.end() - 1);
 
+  // Store result of scatter if needed for reuse in vjp
+  auto get_result = [&]() {
+    switch (reduce_type_) {
+      case Scatter::Max:
+        return scatter_max(values, indices, updates, axes_, stream());
```

Comment:
This looks good but there should be a `default` here. Otherwise it should at least be a compiler warning and possibly undefined behavior as a non-void function has paths without return value.

Reply:
Done @angeloskath